packages/react-native/Libraries/Text/TextInput/Multiline/RCTUITextView.mm

```diff
@@ -307,11 +342,44 @@ - (void)_updatePlaceholder
 
 - (CGRect)caretRectForPosition:(UITextPosition *)position
 {
+  CGRect originalRect = [super caretRectForPosition:position];
+
   if (_caretHidden) {
     return CGRectZero;
   }
 
-  return [super caretRectForPosition:position];
+  if(_caretYOffset != 0) {
+    originalRect.origin.y += _caretYOffset;
+  }
+
+  if(_caretHeight != 0) {
+    originalRect.size.height = _caretHeight;
+  }
+
+  return originalRect;
+}
+
+- (NSArray *)selectionRectsForRange:(UITextRange *)range {
```

Comment:
why is this override needed? I don't understand what it is trying to achieve on top of handling `_caretYOffset` and `_caretHeight` in the method `caretRectForPosition` above.

Reply:
The goal is to avoid the cursor touching the the symbols of the previous line in the second and subsequent lines:
<img width="288" alt="Screenshot 2024-02-26 at 09 42 32" src="https://github.com/facebook/react-native/assets/19835085/e06cf0c1-d46b-40cb-96aa-4a33bcaee954">

 This is the default behavior derived from `caretRectForPosition`. To fix this, I added two new properties `_caretYOffset` and `_caretHeight` to calculate the caret position and caret height, thus preventing the cursor from overlapping the previous line of text. 

To keep the same effect on selection `selectionRectsForRange` needs to be override as well.

Reply:
Can you add a screenshot of what this looks like with a multi-line text selection?

Reply:
Sure, here is a screenshot for multi-line text selection:

<img width="324" alt="Screenshot 2024-04-05 at 13 07 49" src="https://github.com/facebook/react-native/assets/19835085/ee8f08b0-fc0c-44a9-9cc0-84992e0e4944">